lib/sdf-server/src/extract.rs

```diff
@@ -147,6 +148,25 @@ impl FromRequestParts<AppState> for HandlerContext {
     }
 }
 
+#[derive(Deref)]
+pub struct AssetSprayer(pub asset_sprayer::AssetSprayer);
+
+#[async_trait]
+impl FromRequestParts<AppState> for AssetSprayer {
+    type Rejection = (StatusCode, Json<serde_json::Value>);
+
+    async fn from_request_parts(
+        _parts: &mut Parts,
+        state: &AppState,
+    ) -> Result<Self, Self::Rejection> {
+        let asset_sprayer = state
+            .asset_sprayer()
+            .ok_or(not_found_error("openai not configured"))?;
+        // TODO why do we clone? Does this keep the pool around at all?
+        Ok(Self(asset_sprayer.clone()))
```

Comment:
Because it is taken as a reference from the app state:
```rust
pub fn asset_sprayer(&self) -> Option<&AssetSprayer> {
    self.asset_sprayer.as_ref()
}
```

Reply:
oh, this would be a cargo culted comment. removing